### PR TITLE
Documentation for RapidCore.Mongo

### PR DESCRIPTION
**Pull request by pingvinen**
Includes documentation for the public parts of RapidCore.Mongo, i.e.

 - MongoDbConnection
 - MongoManager / IndexAttribute / EntityAttribute

### CODE REVIEW

File: Mongo/Indexes.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+`EnsureIndexes` will scan the given assembly for classes in the given namespace that have been marked with `[RapidCore.Mongo.Entity]`.
+
+Due to the current state of reflection in .NET Core, you have to provide the assembly to scan.
+
+### Note that
+
+- All key indexes are made in **ascending** order.
+- All indexes are made in the background
+- By default indexes are **not** sparse
```
Comment (nover) [inline Mongo/Indexes.md]: Link to mongo documentation regarding sparse indexes

File: Mongo/Indexes.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+Compound indexes
+----------------
+
+If we want to support a query using both our properties, we should add a compound index - i.e. an index with both properties. We can even add the compound index in addition to the individual indexes.
+
+```csharp
+[Entity]
+public class Entity
+{
+    [Index]
```
Comment (nover) [inline Mongo/Indexes.md:95]: Double index attribute in this case?

**Review by nover — COMMENTED** [review]
Overall good but I believe a few amends are needed

File: Mongo/Indexes.md (hunk 2 above)
Comment (pingvinen) [reply Mongo/Indexes.md:95]: Not double. The index attribute is used twice, as we are making an index for the field on its own _and_ using it in a compound index.

File: Mongo/Indexes.md (hunk 1 above)
Comment (pingvinen) [reply Mongo/Indexes.md]: Done

**Review by MartinsRui — APPROVED** [review]
read it through, it was clear to me.

**Pull request merged**